_test/shared/matlab_xunit/xunit/TestSuite.m

```diff
@@ -45,66 +45,68 @@
 %   See also CommandWindowTestRunDisplay, TestCase, TestComponent, TestRunLogger
 
 %   Steven L. Eddins
+%   Modified J. Wilkins 19-01-2021
 %   Copyright 2008-2010 The MathWorks, Inc.
 
 classdef TestSuite < TestComponent
-    
+
     properties (SetAccess = protected)
         TestCaseClasses = containers.Map();
         TestComponents  = {};
         PrintRunInfo    = containers.Map('KeyType','char','ValueType','logical');
     end
-    
+
     methods
-        
+
         function self = TestSuite(name)
             %TestSuite Constructor
             %   suite = TestSuite constructs an empty test suite. suite =
             %   TestSuite(name) constructs a test suite by searching for test
             %   cases defined in an M-file with the specified name.
-            
+
             if nargin >= 1
                 self = TestSuite.fromName(name);
             end
             self.TestCaseClasses  = containers.Map();
         end
-        
-        
+
+
         function [did_pass_out,num_tests_run] = run(self, monitor,num_tests_run)
             %run Execute test cases in test suite
             %   did_pass = suite.run() executes all test cases in the test
             %   suite, returning a logical value indicating whether or not all
             %   test cases passed.
-            
+
             if nargin < 2
                 monitor = CommandWindowTestRunDisplay();
             end
             if nargin<3
                 num_tests_run = 0;
             end
-            
+
             monitor.testComponentStarted(self);
             did_pass = true;
-            
+
             self.setUp();
-            
+
             for k = 1:numel(self.TestComponents)
                 [this_component_passed,num_tests_run] = self.TestComponents{k}.run(monitor,num_tests_run);
-                did_pass = did_pass && this_component_passed;
+                % Logical cast not "necessary" as auto-cast, but for clarity
+                did_pass = logical(did_pass) && this_component_passed;
```

Comment:
What type is `did_pass`? The cast to boolean suggests it's not boolean. If it's a `TestComponent.[passed|skip|failed]` integer a check that `did_pass ~= TestComponent.failed` is clearer and removes an implicit dependce on `TestComponent.failed` being zero.
The same for `this_component_passed`.